ui/app/manager/src/pages/page-alarms.ts

```diff
@@ -0,0 +1,1133 @@
+            // Means a validation failure shouldn't get here
+            console.log("Second if");
+            return;
+        }
+
+        const isUpdate = !!alarm.id;
+        if (!isUpdate) {
+            alarm.realm = manager.getRealm();
+        }
+
+        try {
+            console.log("Try block");
+            action == "update"
+                ? await manager.rest.api.AlarmResource.updateAlarm(alarm.id, alarm)
+                : await manager.rest.api.AlarmResource.createAlarmWithSource(alarm.source,'alarm-page', alarm).then(async (response) => {
+                    if (alarm.alarmAssetLinks.length > 0) {
+                        alarm.alarmAssetLinks.forEach((link) => {
+                            link.id.sentalarmId = response.data.id;
+                            link.id.realm = response.data.realm;
+                        })
+                        await manager.rest.api.AlarmResource.setAssetLinks(alarm.alarmAssetLinks);
+                    }
+                });
+
+
+        } catch (e) {
+            if (isAxiosError(e)) {
+                console.error(
+                    (isUpdate ? "save alarm failed" : "create alarm failed") + ": response = " + e.response.statusText
+                );
+
+                if (e.response.status === 400) {
+                    showSnackbar(
+                        undefined,
+                        i18next.t(isUpdate ? "alarm.saveAlarmFailed" : "alarm.createAlarmFailed"),
+                        i18next.t("dismiss")
+                    );
+                } else if (e.response.status === 403) {
+                    showSnackbar(undefined, i18next.t("alarm.alarmAlreadyExists"));
+                }
+            }
+            throw e; // Throw exception anyhow to handle individual cases
+        } finally {
+            await this._loadData();
+            this.reset();
+        }
+    }
+
+
+
+    protected render() {
+        if (!manager.authenticated) {
+            return html`
+                <or-translate value="notAuthenticated"></or-translate> `;
+        }
+
+        const disabled = false;
+        const readAlarms = manager.hasRole("read:alarms");
+        const writeAlarms = manager.hasRole("write:alarms");
+
+        const readonly = readAlarms && !writeAlarms;
+        const assignOnly = !readAlarms && !writeAlarms;
+        this.config = {assignOnly: assignOnly};
+        return html`
+            <div id="wrapper">
+                <!-- Alarm Specific page -->
+
+                <div id="title" class="${this.creationState || this.alarm ? "hidden" : ""}" style="justify-content: space-between; margin-top: 10px; margin-bottom: 10px">
+                    <div class="${this.creationState || this.alarm ? "hidden" : ""}">
+                        <or-icon icon="bell-outline" style="margin-left: 20px"></or-icon>
+                        <span> ${this.alarm != undefined ? this.alarm.title : i18next.t("alarm.alarm_plural")} </span>
```

Comment:
You can use the `<or-translate>` component instead. We try to practice it nowadays as it's easier to maintain. Scan through this file and apply where necessary.